dbt/adapters/databricks/python_models/python_submissions.py

```diff
@@ -162,3 +162,161 @@ def submit(self, compiled_code: str) -> None:
+            return existing_job_id, False
+
+        response_jobs = self.api_client.workflows.search_by_name(workflow_spec['name'])
+
+        if len(response_jobs) > 1:
+            raise DbtRuntimeError(
+                f"""Multiple jobs found with name {workflow_spec['name']}. Use a unique job
+                name or specify the `existing_job_id` in the workflow_job_config.""")
+
+        if len(response_jobs) == 1:
+            return response_jobs[0]["job_id"], False
+        else:
+            return self.api_client.workflows.create(workflow_spec), True
+
+    def _build_job_permissions(self, job_id, job_grants) -> list:
+        access_control_list = []
+        current_owner, permissions_attribute = self._get_current_job_owner(job_id)
+        access_control_list.append({
+            permissions_attribute: current_owner,
+            'permission_level': 'IS_OWNER',
+        })
+
+        for grant in job_grants.get('view', []):
+            acl_grant = grant.copy()
+            acl_grant.update({
+                'permission_level': 'CAN_VIEW',
+            })
+            access_control_list.append(acl_grant)
+        for grant in job_grants.get('run', []):
+            acl_grant = grant.copy()
+            acl_grant.update({
+                'permission_level': 'CAN_MANAGE_RUN',
+            })
+            access_control_list.append(acl_grant)
+        for grant in job_grants.get('manage', []):
+            acl_grant = grant.copy()
+            acl_grant.update({
+                'permission_level': 'CAN_MANAGE',
+            })
+            access_control_list.append(acl_grant)
+
+        return access_control_list
+
+    def _get_current_job_owner(self, job_id) -> tuple[str, str]:
+        """
+        :return: a tuple of the user id and the ACL attribute it came from ie:
+            [user_name|group_name|service_principal_name]
+            For example: `("mateizaharia@databricks.com", "user_name")`
+        """
+        permissions = self.api_client.workflow_permissions.get(job_id)
+        for principal in permissions.get("access_control_list", []):
+            for permission in principal['all_permissions']:
+                if permission['permission_level'] == 'IS_OWNER' and permission['inherited'] is False:
+                    if principal.get('user_name'):
+                        return principal['user_name'], 'user_name'
+                    elif principal.get('group_name'):
+                        return principal['group_name'], 'group_name'
+                    else:
+                        return principal['service_principal_name'], 'service_principal_name'
+
+        raise DbtRuntimeError(f"Error getting current owner for Databricks workflow.\n {response.content!r}")
+
+    def _get_or_trigger_job_run(self, job_id: str):
+        """
+        :return: the run id
+        """
+        active_runs = self.api_client.job_runs.list_active_runs_for_job(job_id)
+        if len(active_runs) > 0:
```

Comment:
If the workflow is already running, I think we still need to schedule a run after it completes...this is one of the edge-cases that Amy is worried about.  Consider that in the dbt run, the tables that this model depends on may have been updated after that job run started; in order to ensure that downstream tables get those updates, we need to run the job again.

Reply:
Gotcha - that's a good point. That shouldn't be too hard to work in

Reply:
Didn't realize there was an option for that in the Databricks API already - that made things easy. Done.